diksha/diksha.controller.js

```diff
@@ -423,51 +402,73 @@ let generateResponseStructure = (rSt, rsps) => {
     //console.log(JSON.stringify(foo, null, 4));
 
     defer.resolve({
-        responseStructure: rSt
+	    responseStructure: rSt
     });
     return defer.promise;
 }
 
-let doPrebuiltSearch = (requestSkeletons, query) => {
-    let defer = q.defer();
-    let bulkedResponses = {};
-    let bulkedResponsePromises = [];
-    let sectionNames = [];
-    let keys = Object.keys(requestSkeletons);
-    keys.forEach(key => {
-        if (typeof requestSkeletons[key] === 'undefined' || requestSkeletons[key] === null) {} else {
-            reqSkel = requestSkeletons[key];
-            reqSkel.query = query;
-            sectionNames.push(key);
-            bulkedResponsePromises.push(doThoroughSearch(reqSkel));
-        }
-    });
-    q.all(bulkedResponsePromises).then(values => {
-        let responses = {};
-
-        for (let i = 0; i < sectionNames.length; i++) {
-            responses[sectionNames[i]] = values[i].responses.map(response => response.fields);
-        }
-
-        //Object.keys(responses).forEach((key, index) => {
-        //let naam = '/home/admin/' + index;
-        //console.log('writing: ', naam);
-        //fs.writeFileSync(naam, JSON.stringify(responses[key], null, 4));
-        //console.log(index + '>' + key + ': ' + JSON.stringify(responses[key], null, 4))
-
-        //	console.log(index+1, ":", key);
-        //});
-
-        return defer.resolve({
-            responses
-        });
-    }).catch(e => {
-        console.log(e);
-        return defer.reject(e);
-    });
-    return defer.promise;
+let doSectionwiseSearch = (sectionObject) => {
+	    let searchpromise;
+	    let queryObject = {
+	      
+		   "conjuncts" : [] 
+	    }    
+	    for(let key in sectionObject){
+                
+		    if(key !== "compatibilityLevel"){
+    
+			    if(typeof sectionObject[key] === "object") {
```

Comment:
What happens when other JSON type parameters are introduced, apart from `compatibilityLevel`? May be add a comment explaining this check here. You can also add it to this plugin's documentation that we'll be preparing.